### PR TITLE
ready-to-run example for the periodic 2d hills case

### PR DESCRIPTION
**Pull request by slaizet**

### CODE REVIEW

File: examples/2D-hill/input.i3d [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+beta = 2.    ! Refinement parameter (beta)
+
+! Domain
+xlx = 9.             ! Lx (Size of the box in x-direction)
+yly = 3.036              ! Ly (Size of the box in y-direction)
+zlz = 4.5              ! Lz (Size of the box in z-direction)
+
+! Flow parameters
+re  = 1000.           ! nu=1/re (Kinematic Viscosity)
+cpg = F               ! if cpg=T, then re is friction Reynolds number
```
Comment (pbartholomew08) [inline examples/2D-hill/input.i3d:28]: Does this work as a logical value? Does it not need to be `.true.` or `.false.`?

File: src/BC-Periodic-hill.f90 [hunk 2]
```diff
@@ -281,7 +256,7 @@ subroutine hill_flrt (ux,constant)
 
     can=-(constant-ut4)
 
-    if (nrank==0) write(*,*) nrank,'UT',ut4,can
+    !if (nrank==0) write(*,*) nrank,'UT',ut4,can
```
Comment (pbartholomew08) [inline src/BC-Periodic-hill.f90]: delete?

**Review by pbartholomew08 — CHANGES_REQUESTED** [review]

File: examples/2D-hill/input.i3d (hunk 1 above)
Comment (mathrack) [reply examples/2D-hill/input.i3d:28]: 'F' or 'T' are valid values

File: src/BC-Periodic-hill.f90 [hunk 3]
```diff
@@ -177,24 +178,16 @@ subroutine init_hill (ux1,uy1,uz1,ep1,phi1)
...
+          if (istret.ne.0) y=yp(j+xstart(2)-1)-yly*half
+          do i=1,xsize(1)
+             ux1(i,j,k)=ux1(i,j,k)+one-y*y
+             uy1(i,j,k)=0.
+             uz1(i,j,k)=0.
+          enddo
        enddo
     enddo
-
+
```
Comment (mathrack) [inline src/BC-Periodic-hill.f90:196]: fprettify ?

File: src/BC-Periodic-hill.f90 [hunk 4]
```diff
@@ -294,5 +269,106 @@ subroutine hill_flrt (ux,constant)
...
+    use var, ONLY : nzmsize
+
+    implicit none
+
+    real(mytype), intent(in), dimension(xsize(1),xsize(2),xsize(3)) :: ux1, uy1, uz1, ep1
+    real(mytype), intent(in), dimension(xsize(1),xsize(2),xsize(3),numscalar) :: phi1
+    real(mytype), intent(in), dimension(ph1%zst(1):ph1%zen(1),ph1%zst(2):ph1%zen(2),nzmsize,npress) :: pp3
+
+  end subroutine postprocess_hill
+  subroutine visu_hill_init(visu_initialised)
```
Comment (mathrack) [inline src/BC-Periodic-hill.f90:287]: Add some extra lines between the subroutines ?

Comment (slaizet) [reply src/BC-Periodic-hill.f90:287]: done

File: src/BC-Periodic-hill.f90 (hunk 2 above)
Comment (slaizet) [reply src/BC-Periodic-hill.f90]: done

File: src/BC-Periodic-hill.f90 (hunk 3 above)
Comment (slaizet) [reply src/BC-Periodic-hill.f90:196]: more or less done

**Review by pbartholomew08 — APPROVED** [review]
Looks good to me

**Pull request merged**